### PR TITLE
add a message producer client and optimize imports

### PR DESCRIPTION
**Pull request by allanmckenzie**
NB: I also optimised imports so there are a lot of changed files with just import changes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,14 @@ All notable changes to this project will be documented in this file, which follo
 on [Keep a CHANGELOG](http://keepachangelog.com/). This project adheres to
 [Semantic Versioning](http://semver.org/).
 
+## [0.16.0] - 2016-08-11
```
Comment (mapingo) [inline CHANGELOG.md]: Need to change the date to today if releasing today

File: test-utils/test-utils-core/pom.xml [hunk 2]
```diff
@@ -75,7 +83,17 @@
...
         </dependency>
+        <dependency>
+            <groupId>com.jayway.jsonpath</groupId>
+            <artifactId>json-path-assert</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.hamcrest</groupId>
+            <artifactId>hamcrest-core</artifactId>
+        </dependency>
```
Comment (mapingo) [inline test-utils/test-utils-core/pom.xml]: should this be scope of test?

File: test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/matchers/UuidStringMatcher.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import org.hamcrest.Factory;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeMatcher;
+
+public class UuidStringMatcher extends TypeSafeMatcher<String> {
+
+    private static final String UUID_REGEX = "[0-9a-fA-F]{8}(?:-[0-9a-fA-F]{4}){3}-[0-9a-fA-F]{12}";
+
+    @Override
+    protected boolean matchesSafely(String maybeUuid) {
```
Comment (mapingo) [inline test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/matchers/UuidStringMatcher.java]: finals on method arguments

File: test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/MessageProducerClient.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+import org.apache.activemq.ActiveMQConnectionFactory;
+
+public class MessageProducerClient implements AutoCloseable {
+
+    private static final String QUEUE_URI = System.getProperty("queueUri", "tcp://localhost:61616");
+
+    private Session session;
+    private MessageProducer messageProducer;
+
+    public void startProducer(String topicName) {
```
Comment (mapingo) [inline test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/MessageProducerClient.java]: finals on method arguments

File: CHANGELOG.md (hunk 1 above)
Comment (matt-rich) [inline CHANGELOG.md]: Should be 'not released' until we do the release.

Comment (matt-rich) [inline CHANGELOG.md]: Done

File: test-utils/test-utils-core/pom.xml (hunk 2 above)
Comment (matt-rich) [inline test-utils/test-utils-core/pom.xml]: DONe

File: test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/matchers/UuidStringMatcher.java (hunk 3 above)
Comment (matt-rich) [inline test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/matchers/UuidStringMatcher.java]: Done

File: test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/MessageProducerClient.java (hunk 4 above)
Comment (matt-rich) [inline test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/MessageProducerClient.java]: Done

**Comment by y725** [comment]
I would suggest create a common parent class for MessageConsumerClient and MessageProducerClient and move the common code (start connection /close) there just to reduce code  duplication.

Also if you need unit test it there is an embedded ActiveMQ broker (see class org.apache.activemq.broker.BrokerService ).

File: CHANGELOG.md (hunk 1 above)
Comment (purple52) [inline CHANGELOG.md]: FYI, the change log recommends using "[Unreleased]".

**Pull request merged**